modelci/runs.py

```diff
@@ -0,0 +1,308 @@
+        logger.info(f'Container with label: {label} not found')
+
+
+def start_mongodb(docker_client, port=MONGO_PORT):
+    """Start Mongo DB service.
+    From https://stackoverflow.com/a/53522699/13173608.
+
+    Args:
+        docker_client (docker.client.DockerClient): Docker client instance.
+        port (int): Port for MongoDB service.
+    """
+
+    mongo_name = _check_service_started(docker_client, label=MONGO_CONTAINER_LABEL)
+
+    if mongo_name is None:
+        mongo_name = f'mongo-{random.randint(0, 100000)}'
+
+        init_sh_path = str(Path(__file__).parent.absolute() / 'init-mongo.sh')
+
+        docker_client.containers.run(
+            'mongo', detach=True, ports={'27017/tcp': port}, name=mongo_name,
+            volumes={init_sh_path: {'bind': '/docker-entrypoint-initdb.d/init-mongo.sh', 'mode': 'ro'}},
+            environment={
+                'MONGO_INITDB_USERNAME': MONGO_USERNAME,
+                'MONGO_INITDB_PASSWORD': MONGO_PASSWORD,
+                'MONGO_INITDB_DATABASE': MONGO_DB,
+            },
+            labels=[MONGO_CONTAINER_LABEL]
+        )
+
+    check_container_status(docker_client, name=mongo_name)
+    logger.info(f'Container name={mongo_name} stared')
+
+
+def stop_mongodb(docker_client):
+    """Stop Mongo DB service.
+
+    Args:
+        docker_client (docker.client.DockerClient): Docker client instance.
+    """
+    _stop_service(docker_client, MONGO_CONTAINER_LABEL)
+
+
+def start_cadvisor(docker_client, gpu=False, port=CADVISOR_PORT):
+    """Start cAdvisor service.
+
+    Args:
+        docker_client (docker.client.DockerClient): Docker client instance.
+        gpu (bool): Flag for enable GPU.
+        port (int): Port for cAdvisor service.
+    """
+    cadvisor_name = _check_service_started(docker_client, label=CADVISOR_CONTAINER_LABEL)
+
+    if cadvisor_name is None:
+        cadvisor_name = f'cadvisor-{random.randint(0, 100000)}'
+
+        volumes = {
+            '/': {'bind': '/rootfs', 'mode': 'ro'},
+            '/var/run': {'bind': '/var/run', 'mode': 'rw'},
+            '/sys': {'bind': '/sys', 'mode': 'ro'},
+            '/var/lib/docker': {'bind': '/var/lib/docker', 'mode': 'ro'},
+        }
+
+        if gpu:
+            # find libnvidia-ml.so.1
+            cache_file = Path('/tmp/libnvidia-ml.cache')
+            if cache_file.exists():
+                with open(cache_file) as f:
+                    lib_path = f.read().strip()
+            else:
+                args1 = ('locate', 'libnvidia-ml.so.1')
+                args2 = ('grep', '-v', 'lib32')
+                args3 = ('head', '-1')
+                locate = subprocess.Popen(args1, stdout=subprocess.PIPE)  # NOQA: B603
+                grep = subprocess.Popen(args2, stdin=locate.stdout, stdout=subprocess.PIPE)
+                locate.wait()
+                grep.wait()
+                # NOQA: B603
+                lib_path = subprocess.check_output(args3, stdin=grep.stdout, universal_newlines=True).strip()
```

Comment:
change `lib_path` to `libnvidia_ml_path`.

Reply:
refactored